Sources/NIO/ByteBuffer-views.swift

```diff
@@ -187,3 +187,17 @@ extension ByteBuffer {
         self = view._buffer.getSlice(at: view.startIndex, length: view.count)!
     }
 }
+
+extension ByteBufferView: Equatable {
+    /// required by `Equatable`
+    public static func == (lhs: ByteBufferView, rhs: ByteBufferView) -> Bool {
+        lhs._buffer == rhs._buffer
```

Comment:
I don't think this implementation is right, but to explain why we need to consider what it must mean for two ByteBufferViews to be equal.

Conceptually the equality constraint for BBV should be the same as Array: two BBVs are equal if their contents are equal. This patch doesn't achieve that because it disregards the `range`s of each BBV.

We likely need to slice out the relevant ranges from each backing buffer and then compare those for equality instead.

Reply:
Thanks for the review! I see why this couldn't work, I missed out something pretty foundamental.

I'll likely have some time later today to make the suggested changes and push.